ui/src/pages/visualization.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useRef, useState, useEffect } from 'react';
+import ForceGraph3D from 'react-force-graph-3d';
+// import graphData from './output.json';
+import { Link } from 'react-router-dom';
+import { useUmap } from '../queries/useUmapData';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+export interface GraphNode {
+  id: string;
+  name: string;
+  description: string;
+  cell_line: string;
+  assay?: string;
+  x: number;
+  y: number;
+  z: number;
+}
+
+export const Graph3D: React.FC = () => {
+  const { data: umap_data, isLoading, error } = useUmap();
+  const fgRef = useRef<any>();
+  const [searchId, setSearchId] = useState('');
+  const [message, setMessage] = useState('');
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [colorByOption, setColorByOption] = useState<'cell_line' | 'assay'>('cell_line');
```

Comment:
I'd make this a type or enum up there by the `GraphNode` interface:
```ts
type ColorBy = 'cell-line' | 'assay';
```
Then the `useState` declaration looks like:
```tsx
  const [colorByOption, setColorByOption] = useState<ColorBy>('cell-line');
  ```